### PR TITLE
feat: Added recovery handler

### PR DESCRIPTION
**Pull request by flc1125**
## 📑 Description

I can define a general or global recovery that catches system-level exceptions and identifies those that can be reported.

Example: unit test，

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] All the tests have passed

### CODE REVIEW

**Comment by hwbrzzl** [comment]
Thanks for your PR, please add the `Review Ready` label when this PR can be reviewed.

File: support/debug/recovery/panic_handler.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package recovery
+
+import (
+	"github.com/goravel/framework/contracts/support/debug/recovery"
+)
+
+type PanicHandler struct{}
```
Comment (hwbrzzl) [inline support/debug/recovery/panic_handler.go:7]: Q: Can you please give an example for how to use this in goravel/goravel?

Comment (flc1125) [reply support/debug/recovery/panic_handler.go:7]: It involves my another discussion: https://github.com/goravel/goravel/issues/209

My idea is to have a Kernel to start the entire application, and this startup is a root command.

I can configure the recovery from this root command.

Comment (hwbrzzl) [reply support/debug/recovery/panic_handler.go:7]: Okay, let's discuss it at https://github.com/goravel/goravel/issues/209

**Comment by flc1125** [comment]
Since this discussion(https://github.com/goravel/goravel/issues/209) is closed, this one is closed as well. But if there are new ideas, we can always continue to discuss them.

Welcome, everybody.

**Pull request closed without merging**